plugins/pulp_rpm/plugins/profilers/yum.py

```diff
@@ -381,7 +393,16 @@ def _translate_erratum(unit, repo_ids, consumer, conduit):
                         {'key': unit_key, 'repos': repo_ids, 'consumer': consumer.id}
             raise InvalidUnitsRequested(message=error_msg, units=unit_key)
 
-        updated_rpms = YumProfiler._get_rpms_from_errata(errata)
+        # Get rpms from errata
+        errata_rpms = YumProfiler._get_rpms_from_errata(errata)
+
+        # filter out RPMs we don't have access to. This is similar to the check
+        # being done in _is_errata_applicable().
+        updated_rpms = []
+        for errata_rpm in errata_rpms:
+            if YumProfiler._create_nevra(errata_rpm) in available_rpm_nevras:
```

Comment:
`_create_nevra()` returns a dict with the nevra of the unit, in order to make the list of all available RPMs more memory efficient. Things get really slow without this, especially during other applicability calculations.